vacs-client/frontend/src/components/DAKeyArea.tsx

```diff
@@ -1,16 +1,75 @@
 import {useSignalingStore} from "../stores/signaling-store.ts";
 import DAKey from "./ui/DAKey.tsx";
+import Button from "./ui/Button.tsx";
+import {navigate} from "wouter/use-browser-location";
+import {ClientInfoWithAlias} from "../types/client-info.ts";
 
-function DAKeyArea() {
+type DAKeyAreaProps = {
+    filter: string;
+};
+
+function DAKeyArea({filter}: DAKeyAreaProps) {
     const clients = useSignalingStore(state => state.clients);
+    const grouping = useSignalingStore(state => state.getActiveStationsProfileConfig()?.grouping);
 
     return (
         <div className="grid grid-rows-6 grid-flow-col h-full py-3 px-2 gap-3 overflow-x-auto overflow-y-hidden">
-            {clients.map((client, idx) =>
-                <DAKey key={idx} client={client}/>
-            )}
+            {(() => {
+                switch (grouping) {
+                    case "Fir":
+                    case "Airport": {
+                        if (filter !== "") {
+                            return renderClients(clients.filter(client => client.displayName.startsWith(filter)));
+                        }
+
+                        const slice = grouping === "Fir" ? 2 : 4;
+                        return renderGroups(getGroups(clients, slice));
+                    }
+                    case "FirAndAirport": {
+                        if (filter === "") {
+                            return renderGroups(getGroups(clients, 2));
+                        } else if (filter.length === 2) {
+                            return renderGroups(getGroups(clients, 4, filter));
+                        }
+                        return renderClients(clients.filter(client => client.displayName.startsWith(filter)));
+                    }
+                    case undefined:
+                    case "None":
+                    default:
+                        return renderClients(clients);
+                }
+            })()}
         </div>
     );
 }
 
+function getGroups(clients: ClientInfoWithAlias[], slice: number, prefix = "") {
+    return [
+        ...clients
+            .filter(client => client.displayName.startsWith(prefix))
+            .reduce<Set<string>>((acc, val) =>
+                acc.add(val.displayName.slice(0, slice)
```

Comment:
We should potentially split by `_` and take up to `slice` characters, otherwise the `Airport` split will give us a `LON_` button for our colleagues from the UK.